text/0000-create-stackify-repo.md

```diff
@@ -0,0 +1,141 @@
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+Create a new repo called “stackify” which can be used to turn OCI images into CNB compliant stack images.
+
+# Definitions
+[definitions]: #definitions
+N/A
+
+# Motivation
+[motivation]: #motivation
+Currently, for users to extend the published stack images, or turn their own images into stack images, they have to create Dockerfiles which take the base images, apply their changes, and then set the CNB metadata. It can be difficult to generate and set all of the relevant metadata, especially the mixins.
+
+Stackify will be a tool that they can use to perform all of these manipulations to their base image without Dockerfiles.
+
+
+# What it is
+[what-it-is]: #what-it-is
+Stackify will be a tool that will be useful to users who would like to customize the published stack images or turn their own base images into CNB compliant stack images.
+
+Users can use stackify to add packages and ca-certificates to a base image as well as add all required CNB metadata to the image.
+
+Stackify can be integrated into Pack in the form of a `pack create-stack` command.
+
+# How it Works
+[how-it-works]: #how-it-works
+For stacks defined by the buildpacks project, Stackify will add mixins labels for each package installed on the image. If Stackify is used to give add a stack ID other than one defined by the buildpacks project, mixin labels will not be added.
+
+If Stackify is used with an already valid stack image, the `CNB_*` labels and users will not be changed unless new user and group ID's are passed, in which case the labels and user will be overwritten.
+
+If run on the image being "stackify-ed", Stackify can be used to add packages and ca-certificates.
+
+Stackify will use Kaniko to create additional image layers with the additional packages and ca-certificates that users choose to add to their base image. It will also use GGCR to manipulate the image metadata.
+
+
+### Proposed Interface
+Stackify will be run once for each image (build and run):
+```
+stackify 
+  --base-image string
+    	Source image ref
+  --stack-image string
+    	Destination for stack image
+  --certs-dir string
+    	Path to directory with CA Certs to add to image
+  --group-id string
+    	CNB Group ID
+  --image-type string
+    	Type of image to create (build or run)
+  --mixins string
+    	Comma-separated list of mixins to add to labels
+  --packages string
+    	Comma-separated list of packages to add to image
+  --stack-id string
+      Stack ID to add to image metadata
+  --user-id string
+    	CNB User ID
+```
+
+This will be called under the hood by the `pack create-stack` command, which will take a TOML config file:
+
+`pack create-stack my-stack -c /path/to/config.toml`
+
+Config file:
+```
+[build]
+image = "paketobuildpacks/build:base"
+outputTag = "some-registry/some-custom-stack-build-image"
```

Comment:
For consistency with `create-builder` and `package-buildpack` I would prefer that users must pass the destination tags on the command line.

For example a user could run `pack create-stack` with `--run-image <tag>` `--build-image <tag>` or both. This has an added benefit in that users may not always want to build both images. If for example, I am rebuilding a build image because of a vulnerability in a package that exists only on the run image, I may not want to wait for an identical copy of the run image to build.

Reply:
@ekcasey I agree with the second point (creating a functionality for users to recreate either the build or run image, but not always both), but I'm not sure I feel the first point. Since users are already going to be passing in a config file, I'm not sure I get the utility of having them also pass information in through flags.

Reply:
I agree that we should allow building just one of the stack images. For consistency with `create-builder` and `package-buildpack` we'd be ok passing those tags as flags.

Reply:
@dfreilich If we wanted to allow users to specify the destination tag in the file, I think we should do that for `pack builder create` and `pack buildpack package` as well. I think that discussion should be separate from this RFC (in a `pack` issue?) and we should make the change everywhere simultaneously to keep things as consistent and predictable as possible for users.

I.E I think this RFC only needs to include usage that is consistent with the current `pack` conventions. If we change the `pack` conventions then either:
1. This RFC has already been implemented and we change the usage at the same time we change that of the other commands
2. This RFC has not yet been implemented, but it is okay that the usage in the RFC is slightly different than what gets implemented due to `pack` changes in the interim.